src/components/heading/heading.component.tsx

```diff
@@ -19,9 +20,40 @@ import {
 } from "./heading.style";
 import useLocale from "../../hooks/__internal__/useLocale";
 
-const marginPropTypes = filterStyledSystemMarginProps(
-  styledSystemPropTypes.space
-);
+export interface HeadingProps extends MarginProps, TagProps {
+  /** Child elements */
+  children?: React.ReactNode;
+  /** Defines the title for the heading. */
+  title?: React.ReactNode;
+  /** Defines the title id for the heading. */
+  titleId?: string;
+  /** Defines the subheader for the heading. */
+  subheader?: React.ReactNode;
+  /** Defines the subtitle id for the heading. */
+  subtitleId?: string;
+  /** Defines the help text for the heading. */
+  help?: string;
+  /** Defines the help link for the heading. */
+  helpLink?: string;
+  /** Defines the a href for the back link. */
+  backLink?:
+    | string
+    | ((
+        ev:
+          | React.MouseEvent<HTMLAnchorElement>
+          | React.MouseEvent<HTMLButtonElement>
+          | React.KeyboardEvent<HTMLAnchorElement>
+          | React.KeyboardEvent<HTMLButtonElement>
+      ) => void);
+  /** Adds a divider below the heading and the content. */
+  divider?: boolean;
+  /** Adds a separator between the title and the subheader. */
+  separator?: boolean;
+  /** Pills that will be added after the title. */
+  pills?: React.ReactNode;
+  /** Aria label for rendered help component */
+  helpAriaLabel?: string;
+}
 
 const Heading = ({
```

Comment:
```suggestion
export const Heading = ({
```

I'm not sure why, but after converting the components to TS the props don't show up properly in the Storybook docs unless the component is also explicitly exported.